canonicalwebteam/discourse/parsers/docs.py

```diff
@@ -507,11 +514,10 @@ def _generate_navigation(self, navigations, version_path):
                 item["navlink_href"]
             ):
                 topic_id = self._get_url_topic_id(item["navlink_href"])
-
                 if topic_id in self.url_map_versions[version_path]:
-                    item["navlink_href"] = self.url_map_versions[version_path][
-                        topic_id
-                    ]
+                    href = self.url_map_versions[version_path][topic_id]
+                    fragment = item["navlink_fragment"]
+                    item["navlink_href"] = f"{href}#{fragment}"
```

Comment:
This concatenation is adding `#` at the end of all the URLs. Could it be only when a fragment exists?